src/compressed_tensors/compressors/model_compressor.py

```diff
@@ -90,45 +98,51 @@ def from_pretrained(
         configs and load a ModelCompressor
 
         :param pretrained_model_name_or_path: path to model config on disk or HF hub
-        :return: compressor for the extracted configs
+        :return: compressor for the configs, or None if model is not compressed
         """
         config = AutoConfig.from_pretrained(pretrained_model_name_or_path, **kwargs)
         compression_config = getattr(config, COMPRESSION_CONFIG_NAME, None)
         return cls.from_compression_config(compression_config)
 
     @classmethod
-    def from_compression_config(cls, compression_config: Dict[str, Any]):
+    def from_compression_config(
+        cls, compression_config: Union[Dict[str, Any], "CompressedTensorsConfig"]
+    ):
         """
-        :param compression_config: compression/quantization config dictionary
-            found under key "quantization_config" in HF model config
-        :return: compressor for the extracted configs
+        :param compression_config:
+            A compression or quantization config
+
+            The type is one of the following:
+            1. A Dict found under either "quantization_config" or "compression_config"
+                keys in the config.json
+            2. A CompressedTensorsConfig found under key "quantization_config" in HF
+                model config
+        :return: compressor for the configs, or None if model is not compressed
         """
         if compression_config is None:
             return None
 
-        try:
-            from transformers.utils.quantization_config import CompressedTensorsConfig
+        # check for CompressedTensorsConfig
+        if is_compressed_tensors_config(compression_config):
+            compression_config = compression_config.to_dict()
+            sparsity_config = compression_config["sparsity_config"]
+            quantization_config = compression_config["quantization_config"]
 
-            if isinstance(compression_config, CompressedTensorsConfig):
-                compression_config = compression_config.to_dict()
-        except ImportError:
-            pass
+        else:
+            # parse from dict
+            sparsity_config = cls.parse_sparsity_config(compression_config)
+            quantization_config = cls.parse_quantization_config(compression_config)
```

Comment:
Can we call `cls.parse_sparsity_config()` and `cls.parse_quantization_config()` for both cases, since each of the parsing functions also checks `is_compressed_tensors_config`?

Reply:
Yep makes sense